src/runtime/vulkan/vulkan_device_api.h

```diff
@@ -106,7 +106,7 @@ class VulkanDeviceAPI final : public DeviceAPI {
    * Returns the results of feature/property queries done during the
    * device initialization.
    */
-  void GetTargetProperty(Device dev, const std::string& property, TVMRetValue* rv);
+  void GetTargetProperty(Device dev, const std::string& property, TVMRetValue* rv) final;
```

Comment:
A bit curious about why adding `final` here?

Reply:
Not necessary, but I try to be in the habit of including it for virtual functions that aren't intended to be subclassed further.  It's always safe to remove `final` later, but until then it serves as a warning to my future self that I should be look into the class's implementation in detail before making any subclass of it.

I can go either way on having it or not, since `final` could be interpreted either as "not intended to be subclassed, so take a careful look first" or as "intended not to be subclassed, please don't change it".  I tend to take the first interpretation, but the second is completely valid.

Reply:
I see, thanks!